ref-exchange/src/pool.rs

```diff
@@ -3,55 +3,88 @@ use near_sdk::{AccountId, Balance};
 
 use crate::admin_fee::AdminFees;
 use crate::simple_pool::SimplePool;
+use crate::stable_swap::StableSwapPool;
 use crate::utils::SwapVolume;
 
 /// Generic Pool, providing wrapper around different implementations of swap pools.
 /// Allows to add new types of pools just by adding extra item in the enum without needing to migrate the storage.
 #[derive(BorshSerialize, BorshDeserialize)]
 pub enum Pool {
     SimplePool(SimplePool),
+    StableSwapPool(StableSwapPool),
 }
 
 impl Pool {
     /// Returns pool kind.
     pub fn kind(&self) -> String {
         match self {
             Pool::SimplePool(_) => "SIMPLE_POOL".to_string(),
+            Pool::StableSwapPool(_) => "STABLE_SWAP".to_string(),
         }
     }
 
     /// Returns which tokens are in the underlying pool.
     pub fn tokens(&self) -> &[AccountId] {
         match self {
             Pool::SimplePool(pool) => pool.tokens(),
+            Pool::StableSwapPool(pool) => pool.tokens(),
         }
     }
 
     /// Adds liquidity into underlying pool.
     /// Updates amounts to amount kept in the pool.
-    #[allow(unused_variables)]
     pub fn add_liquidity(
-        &mut self, 
-        sender_id: &AccountId, 
-        amounts: &mut Vec<Balance>, 
-        fees: AdminFees,
```

Comment:
Would it be good to keep a single `add_liquidity` method instead of `add_liquidity` and `add_stable_liquidity` which looks more consistent if we're adding more pools in future.

Reply:
Sure it would be good to combine these two method. But for now, the interface signatures are different, cause the lp tokens in these 2 types of pools has different meanings. For example, in stable pool user cares about how much lp token he would get so the slippage is on min_shares but a simple pool doesn't. additionally, the frontend can not mix these 2 type of pools up for their own reasons, it is nature to call different interface from frontend.

Reply:
For a stable swap pool with specific tokens, the lp shares minted is close to the sum amount of deposit tokens when add liquidity. And the fee is charged in form of reducing some lp shares user would get. So it is nature to make slippage on it. As we can get lp token price easily, if a user care more about pure price or total value they get, the FE can help him to get that infomation.